lib/syskit/exceptions.rb

```diff
@@ -475,24 +475,24 @@ def initialize(deployment_to_tasks, toplevel_tasks_to_requirements = {})
         end
 
         def pretty_print(pp)
-            deployment_to_tasks.each do |deployed_task, tasks|
-                process_server_name = deployed_task.configured_deployment
-                                                   .process_server_name
-                orogen_model = deployed_task.configured_deployment.orogen_model
+            deployment_to_tasks.each do |orocos_name, tasks|
+                agent = tasks.first.execution_agent
+                deployment_m = agent.deployed_orogen_model_by_name(orocos_name)
```

Comment:
You need to extract this information in the constructor. There is no guarantee that the plan will be the way it was once pretty_print is called